### PR TITLE
proxysql 1.4.11 (new formula)

### PR DESCRIPTION
**Pull request by zbentley**
Add formula for Proxysql (sysown/proxysql) with a few patches
to support the most recent release, all of which are either fixed
or pending-apporoval in an upcoming release.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/proxysql.rb
Comment (commitay) [inline Formula/proxysql.rb]: ❓

Comment (zbentley) [reply Formula/proxysql.rb]: fixing

Comment (zbentley) [reply Formula/proxysql.rb]: had a symlink in place instead of the actual formula, sorry. uploading real deal momentarily.

File: Formula/proxysql.rb
Comment (commitay) [inline Formula/proxysql.rb]: Please open an issue upstream requesting a install target with a configurable prefix.

File: Formula/proxysql.rb
Comment (commitay) [inline Formula/proxysql.rb]: The test needs to exercise and verify the functionality of the software.

File: Formula/proxysql.rb
Comment (commitay) [inline Formula/proxysql.rb]: This needs to be accepted by upstream first and we'll probably want to wait for a new release so the patches aren't needed.

Comment (zbentley) [reply Formula/proxysql.rb]: Would a compromise be acceptable here? If upstream accepts that PR but does not cut a release, could we ship the formula backporting the change until the release is cut? I promise that, unless I lose my job and access to a computer, I'll update the formula as soon as .10 is cut. 

I have a moderate number of users for which installing a custom formula will be confusing. That's the only reason I ask this.

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: The issue that I applied patches for (link time problem) is exercised by execution, --help or not, alone.

Otherwise, exercising this program requires:
- Installing mysql
- Starting a database server
- Creating databases/schemas on that server that can be used
- Starting a daemon installed by this package
- Connecting to the databases via that daemon
- Performing operations on them
- Cleaning up

That seems a bit fragile. Should I write a bash/some-other-language program to test this behavior? Otherwise, what would you recommend? Happy to do whatever makes the most sense.

Comment (zbentley) [reply Formula/proxysql.rb]: My main concern is pre-existing installs of `proxysql` or `mysql` (very common) with the same addressing (e.g. DB name, user name) as whatever tests I create. If I collide with those, what should happen?

**Comment by commitay** [comment]
Build failure on el capitain
https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/26141/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_proxysql/

**Comment by zbentley** [comment]
Will investigate El Cap build failure tomorrow. Have to sleep now.

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: Improved tests. Are these sufficient?

Addressing your other comments soon.

**Comment by zbentley** [comment]
Currently trying to diagnose a build failure for this:
```
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1045 (28000): ProxySQL Error: Access denied for user 'admin'@'' (using password: NO)
```

The first line comes from the client, the second from the server (proxysql). It's quite bizarre that the server doesn't seem to be detecting the supplied password (which the commandline client is warning about being supplied). I'll try fiddling with auth protocols in case the `mysql` version installed by Homebrew is too new for the authentication protocols supported by proxysql. If that's the case, I can either override auth protocols on the commandline, or have a test dependency on e.g. `mysql@5.5`, which I suspect is frowned-upon, but am not sure.

Will update with progress.

**Comment by zbeekman** [comment]
@BrewTestBot test this please

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: Patches accepted.

File: Formula/proxysql.rb
Comment (DomT4) [inline Formula/proxysql.rb]: macOS ships `make`, this should be redundant.

File: Formula/proxysql.rb
Comment (DomT4) [inline Formula/proxysql.rb]: Is there no support for the latest `mysql`?

File: Formula/proxysql.rb
Comment (DomT4) [inline Formula/proxysql.rb]: We might well want to wait for upstream to do something about this, given the issue seems to have been ignored thus far. Would prefer not to get stuck with manual `.install` calls indefinitely.

File: Formula/proxysql.rb
Comment (DomT4) [inline Formula/proxysql.rb]: I noticed upstream bundles a large chunk of its own deps, including `openssl`. Is it actually using our `openssl` or the bundled static one or both or?

File: Formula/proxysql.rb
Comment (DomT4) [inline Formula/proxysql.rb]: We should use Ruby's `sleep` rather than shelling out there. This also isn't quite how `pipe_output()` is structured FWIW; normal usage is like [this](https://github.com/Homebrew/homebrew-core/blob/ff464a42eeb4190d1c2f449a8acb19929e256acd/Formula/gocryptfs.rb#L31).

**Review by DomT4 — CHANGES_REQUESTED** [review]

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: Unfortunately the upstream is in an unfortunate spot of having a) not many maintainers, b) a very opinionated build system, and c) the assumption about bundled libraries and fixed paths baked into the build process at multiple points.

I'd love to see it change, but I do not consider it likely, especially given that Renée &co have a ton of other stuff they're working on feature-wise to get 2.0.0 released.

On the bright side, these manual `install` calls are a bit less sketchy than they could be, since the proxysql app compiles in all of its dependencies, and tries very hard to make its deployable a single filesystem-location-independent binary. As a result, I think this is more of the "ugly" variety of `install` hacks and  less of the "potential breakage" variety. What do you think?

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: 👍 will fix.

File: Formula/proxysql.rb
Comment (javian) [inline Formula/proxysql.rb]: Wouldn't you want to add a plist method so that it can be run as a service ?

File: Formula/proxysql.rb
Comment (DomT4) [reply Formula/proxysql.rb]: > If you'd like me to spend more time on that, I'm happy to--just let me know.

No need. I'd prefer this to always use our OpenSSL, so we're not waiting on upstream to catch up to security updates.

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: Tests are fixed now on normal mysql.

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: 👍

File: Formula/proxysql.rb
Comment (zbentley) [reply Formula/proxysql.rb]: 👍

**Comment by zbentley** [comment]
@BrewTestBot test this please

**Comment by zbentley** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
I see that upstream is approaching a 2.0.0 release. We'll probably wait for that, and hopefully it can incorporate a sane building system or `make install` method.

**Comment by zbentley** [comment]
@fxcoudert that does not seem likely; there has been no discussion on the issue or chats/lists about changing the build system by 2.0.0 release.

If this formula is updated to point at 2.0.0 when they leave RC, could it be merged? I don't want to keep bumping it/committing if it's unlikely to be accepted; no problems either way.

**Pull request closed without merging**
